### PR TITLE
Sticky team selections: Chunk 1, Frontend amazingness

### PR DESCRIPTION
**Pull request by ticky**
This pull request persists the last selected* team for a given Organization to a user’s localStorage, and updates the Navigation link to match.

- [x] 1. Create a localStorage-based store technology for session key-values
- [x] 2. Push last seen team selection on pipelines page into 1
  - [x] If page loads with a selected team, update key-value store with team slug
  - [x] If changing the team in the UI, update the store with the team slug
- [x] 3. Listen to store in navigation, update pipelines page link with team slug

_*this means it can get inconsistent when switching tabs, which is a bit weird, but hey_

### CODE REVIEW

File: app/components/layout/Navigation/index.js [hunk 1]
```diff
@@ -26,13 +28,43 @@ class Navigation extends React.Component {
 
   componentDidMount() {
     this.props.relay.setVariables({ isMounted: true });
+    SessionHashStore.on('change', this.handleSessionDataChange);
+  }
+
+  componentWillMount() {
+    if (this.props.organization && this.props.organization.slug) {
+      this.setState({
+        lastDefaultTeam: SessionHashStore.get(`organization-default-team:${this.props.organization.slug}`)
```
Comment (keithpitt) [inline app/components/layout/Navigation/index.js]: Perhaps we should use the `id` here to guard against name changes?

File: app/components/organization/Pipelines.js [hunk 2]
```diff
@@ -58,6 +62,20 @@ class OrganizationPipelines extends React.Component {
         });
       });
     }
+
+    this.maybeUpdateDefaultTeam(nextProps.organization.slug, nextProps.team);
+  }
+
+  maybeUpdateDefaultTeam(organization, team) {
```
Comment (keithpitt) [inline app/components/organization/Pipelines.js]: Missed opportunity: `callMeMaybeUpdateDefaultTeam`

**Comment by keithpitt** [comment]
I am super excited about this technology!

File: app/components/organization/Pipelines.js (hunk 2 above)
Comment (ticky) [inline app/components/organization/Pipelines.js]: Didn’t I already make that joke in another PR? 😜

File: app/components/layout/Navigation/index.js (hunk 1 above)
Comment (ticky) [inline app/components/layout/Navigation/index.js]: The `id` of the org or the team?

File: app/components/organization/Pipelines.js (hunk 2 above)
Comment (keithpitt) [inline app/components/organization/Pipelines.js]: Maybe?

File: app/components/layout/Navigation/index.js (hunk 1 above)
Comment (keithpitt) [inline app/components/layout/Navigation/index.js]: The `id` of the org

**Comment by keithpitt** [comment]
:rocket:
